flink-core/src/main/java/org/apache/flink/core/memory/HybridMemorySegment.java

```diff
@@ -57,6 +59,10 @@
      */
     @Nullable private ByteBuffer offHeapBuffer;
 
+    @Nullable private final Runnable cleaner;
+
+    private final boolean allowWrap;
```

Comment:
It might not be super clear which segments are allowed to be wrapped and which not. Adding some explanation could help.